### DISCUSSION TITLE
Realizzazione progetto Sonda di assorbimento

### DISCUSSION THREAD

**Discussion by Robyn68** [indomus/forum#90] [Hardware & Firmware]
Ciao a tutti.
Mi sono messo in testa di realizzare il progetto riportato in questo link: https://indomus.it/progetti/realizzare-una-sonda-di-assorbimento-elettrico-domotica-tramite-pzem-e-esp8266-wemos-nodemcu-o-altri/ ...
Mi chiedevo se chi lo aveva redatto o chi lo aveva già realizzato, potesse darmi qualche indicazione in più sulle 2 resistenze necessarie per il progetto ( Watt, tolleranza, possibilità di utilizzare anche altri valori vicini ai 2k).
Grazie a tutti.

**Comment by gio-dot** [comment]
Che resistori hai? Già ti dico che Watt e tolleranza sono del tutto indifferenti. Hai già il pzem e il Wemos/node mcu?

**Reply by Robyn68** [reply]
Ciao.
I resistori che più si avvicinano ai valori indicati sono 1,2k e 3,3k.
Si, ho già sia Pzem che D1 mini.

**Reply by gio-dot** [reply]
Puoi usare 3k3. Se ti interessa puoi vedere anche qui per un progetto più avanzato:
https://github.com/Gio-dot/PZEM-016-OLED-2-OUT-ESPHome

**Reply by Robyn68** [reply]
Grazie mille!
Appena ho un attimo ci guardo

**Reply by Robyn68** [reply]
Bella cosa l'aggiunta del display...
Due cose:
- Non capisco perché in quel progetto si preferisca prendere la tensione per alimentare il D1 direttamente ai capi di c4, piuttosto che dai morsetti.
- Altra cosa, è necessario rimuovere U4 e R10, ma quest'ultima non ce l'ho sulla mia board.

**Reply by gio-dot** [reply]
Perchè il D1 è messo dentro al PZEM e così facendo non ho fili cha vanno da fuori a dentro.
Per la R10 non so che dire, forse hannp modificato la board.

**Reply by gio-dot** [reply]
Altra alternativa se ti può interessare è di farlo zigbee, come ho fatto qui:
https://gio-dot.github.io/Z1-Mini/pzem-zigbee-meter

**Comment by Robyn68** [comment]
Ciao a tutti.
Sono finalmente riuscito ad avere un po' di tempo e ho realizzato il progetto del mio primo post. Se funziona tutto penso all'espansione con anche il display che mi sembra molto figo.. 
Accendo per la prima volta il prototipo e tutto sembra funzionare. Non ho ancora collegato il toroide intorno al cavo in cui passa la fase, ma decido comunque di accenderlo.
Nella homepage del D1 ottengo la rappresentazione dei valori che arrivano dal Pzem tutti a 0 tranne la tensione e la frequenza che immagino li prenda direttamente dall'alimentazione. Mi dico...funziona!!!
Bene, provo a vedere se i dati arrivano anche su HA, anche se sono a 0, torno sulla homepage del D1 e stranamente ora anche i valori di tensione e frequenza sono a 0.
Ho paura sia successo qualcosa. Potete darmi qualche indicazione?
Grazie a tutti,

**Reply by Robyn68** [reply]
da 0 a questo
![image](https://user-images.githubusercontent.com/75576198/114315103-a294db80-9afd-11eb-8d3a-577237d1416d.png)

**Reply by gio-dot** [reply]
Il toroide anche se non ci passa nulla tienilo comunque attaccato al pzem.

**Reply by Robyn68** [reply]
è attaccato...tutto è collegato correttamente e le saldature sono pulite e i contatti non sono in corto...li ho provati uno per uno... sono dipserato...

**Reply by gio-dot** [reply]
Ma ora che vedi sull'interfaccia? Dati fissi o cambiano?

**Reply by Robyn68** [reply]
Continuano a cambiare. Ma sono, come puoi vedere, valori non corretti

**Reply by Robyn68** [reply]
Ho provato con un secondo Pzem e un nuovo D1. Alla partenza tutto ok, poi appena installato mi ha cominciato a dare tutti i valori a 0 compresi frequenza e tensione che ovviamente prende dalla sua alimentazione e non dal toroide... Ma a tutti è andato bene al primo colpo??????
Mi sta venendo il dubbio che modificando il valore delle resistenze, qualcosa ne comprometta il funzionamento... altrimenti non me lo spiego, i dispositivi erano nuovi.

**Comment by sky7176** [comment]
> 
> 
> Ciao a tutti.
> Mi sono messo in testa di realizzare il progetto riportato in questo link: https://indomus.it/progetti/realizzare-una-sonda-di-assorbimento-elettrico-domotica-tramite-pzem-e-esp8266-wemos-nodemcu-o-altri/ ...
> Mi chiedevo se chi lo aveva redatto o chi lo aveva già realizzato, potesse darmi qualche indicazione in più sulle 2 resistenze necessarie per il progetto ( Watt, tolleranza, possibilità di utilizzare anche altri valori vicini ai 2k).
> Grazie a tutti.

Ciao, io al posto di rimuovere l'integrato ho utilizzato un convertitore RS485 - TTL, ho messo il convertitore e lo ESP8266 in una scatolina esterna.
In questo modo non "manometto" il PZEM-016, non è farina del mio sacco, ma ho trovato il tutto su internet.
Devo dire che per chi non è pratico nel dissaldare componenti SMD forse è una buona alternativa.

Giuseppe

**Comment by array81** [comment]
Mi riallaccio a questo topic anche se vecchio per non aprirne un altro.
Intanto giusto a titolo informativo il componente SMD può non essere dissaldato. Basta dissaldare un solo piedino (informazione reperita in rete e testata con successo).

Detto questo voglio realizzare una seconda sonda per misurare la produzione del mio impianto fotovoltaico. Insieme al PZEM-016 vendono anche la pinza ma questa ha i cavi lunghi circa 20cm mentre in questo caso mi occorrono cavi lunghi circa 1m perchè devo installare il PZEM lontano da cavo su cui poi andrò ad ancorare la pinza.
Potrei fare una giunta ma l'idea non mi piace molto, l'alternativa è trovare una pinza che abbia i cavi già lunghi. Il problema è che non ho idea delle caratteristiche che deve are la pinza.
Avrei trovato questa: https://it.aliexpress.com/item/1005002007179360.html?spm=a2g0o.cart.0.0.145718fcPLnKBb&mp=1&gatewayAdapt=glo2ita

Potrei tagliare lo spinotto e collegarla al PZEM sono tutte 1V ma cambiano gli Ampere, mi sapete dire se va bene ed eventualmente quale ordinare?
